File: 백준/Gold/1644. 소수의 연속합/소수의 연속합.py
import sys
input = sys.stdin.readline

n = int(input())

def get_prime():
    is_prime = [True] * (n+1)
    prime_list = []
    
    for i in range(2, n+1):
        if is_prime[i]:
            prime_list.append(i)
            j = 2
            while i*j <= n:
                is_prime[i*j] = False
                j += 1
    return prime_list

def get_prime_sum():
    global answer
    start = 0
    end = 1
    prime_sum = prime[0] + prime[1]
    while start < end:
        if prime_sum < n:
            end += 1
            prime_sum += prime[end]
        elif prime_sum > n:
            prime_sum -= prime[start]
            start += 1
        else:
            answer += 1
            end += 1
            prime_sum = prime_sum + prime[end] - prime[start]
            start += 1 

prime = get_prime()

answer = 0
if n > 2:
    get_prime_sum()

if n in prime:
    answer += 1
print(answer)

File: 백준/Gold/1644. 소수의 연속합/소수의 연속합.js
const input = require('fs').readFileSync(process.platform === 'linux' ? '/dev/stdin' : __dirname + '/input.txt').toString().trim().split('\n')

const N = Number(input)
const prime = get_prime()

let answer = prime.find((number) => number === N) ? 1 : 0
let left = 0
let right = 1
let sum = prime[0] + (prime[1] || 0)
while(left < right && right < prime.length) {
  if (sum === N) answer++

  if (sum <= N) {
    right++
    sum += prime[right]
  } else {
    sum -= prime[left]
    left++
  } 
}

console.log(answer)

function get_prime() {
  const prime = Array.from({ length: N + 1 }, () => true)
  prime[1] = false

  const MAX = Math.floor(Math.sqrt(N))
  for (let i = 2; i < MAX + 1; i++){
    if (!prime[i]) continue
    for (let j = i * i; j < N + 1; j += i) {
      prime[j] = false
    }
  }

  const result = []
  for (let i = 1; i < N + 1; i++) {
    if (prime[i]) result.push(i)
  }

  return result
}